### DISCUSSION TITLE
Error: P1001: Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com:6543`

### DISCUSSION THREAD

**Discussion by andreasobi** [prisma/prisma#24477] [Q&A]
### Question

It was working a few days ago and I did not change anything. Supabase just restored the database in the meantime.

### How to reproduce (optional)

1. `npx prisma db push`
2. Getting an error: 

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "aws-0-eu-central-1.pooler.supabase.com:6543"

Error: P1001: Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com:6543`

Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com:6543`._
```

### Information about Prisma Schema, Client Queries and Environment (optional)
.env file:
```
DATABASE_URL="postgres://postgres.xjetlyrodrrupwokmxsa:{password}@aws-0-eu-central-1.pooler.supabase.com:6543/postgres?pgbouncer=true"
DIRECT_URL="postgres://postgres.xjetlyrodrrupwokmxsa:{password}@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"
```
schema.prisma
```
datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id String @id @unique
  email String
  firstName String
  lastName String
  profileImage String
}

model Product {
  id String @id @default(uuid())
  name String
  description String
  createdAt DateTime @default(now())
}

model Lead {
  email String @id
  createdAt DateTime @default(now())
}
```

- macos
- PostGres
- Nextjs 14.1.4
Prisma                  : 5.14.0
@prisma/client          : 5.14.0

**Comment by janpio** [comment]
Thanks for providing all this information!

The output you shared, and your project configuration are not fully consistent. `DIRECT_URL` uses post `5432`, your schema uses `directUrl = env("DIRECT_URL")`, and `db push` would use that - but the output mentions port `6543`. Are you sure you shared the correct information?

> Supabase just restored the database in the meantime.

What do you mean by that? Was that an automated process, or something manually done by them?

Can you manually connect to one of the connection strings with another tool like DBeaver or TablePlus? If so, does that make Prisma work afterwards as well?

If you want and are ok with that, you can shared your full connection strings with me via Discord at username `janprisma` - then I could manually try to verify and debug what is happening to you. I understand of course if that is not possible, then we can try to proceed here.

**Comment by andreasobi** [comment]
Hi Janpio

Yes, I shared the correct information - I just checked again.

<img width="1006" alt="image" src="https://github.com/prisma/prisma/assets/47380490/9f5e98f1-180b-443d-b474-ceb2bfcf9e94">

In Supabase free tier will automatically pause the project after one week of inactivity. This happened to me and then it automatically restored the database after first use. I don't know what Supabase did in the background but it took some time to restore all tables.

I just sent you the connection strings in Discord.

Thanks for the help!

**Reply by janpio** [reply]
> In Supabase free tier will automatically pause the project after one week of inactivity. This happened to me and then it automatically restored the database after first use. I don't know what Supabase did in the background but it took some time to restore all tables.

Ok, free tier pause makes sense. I'll try to incorporate that into my reproduction attempts.



> Yes, I shared the correct information - I just checked again.

When I copy the exact `.env` and `schema.prisma` from above int oa new project, this is what happens on `npx prisma db push`:
```
> npx prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "aws-0-eu-central-1.pooler.supabase.com:5432"

Error: P1001: Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com:5432`

Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com:5432`.
```

Note that it is not working, but it uses port `5432` from the `directUrl`/`DIRECT_URL` - not the `6543` your screenshots show. That makes me a bit curious how this mismatch is happening for you as it could potentially also point to other problem.

**Reply by janpio** [reply]
Bonus question: When you go to the Table editor in the Supabase admin UI for this project, does it show the existing tables?

**Reply by andreasobi** [reply]
I was also wondering about the mismatch... Not sure if this helps.. This is the config in Supabase.
<img width="829" alt="image" src="https://github.com/prisma/prisma/assets/47380490/2478f62f-24e3-4e64-8d0b-324b9b5cb9e3">


Yes, at least from before my project has been paused. No change has been made since then.

<img width="247" alt="image" src="https://github.com/prisma/prisma/assets/47380490/fb8fdedf-52df-4b9a-951e-4733cea2a6ca">

FYI: I sent you a DM (andreasobi) on discord with the connection strings.

**Reply by janpio** [reply]
> I was also wondering about the mismatch...

That should come from something being different than what you shared above. So `.env` file or `schema.prisma`. Maybe there is also an environment variable defined on your actual computer, which would prevent the `.env` from being loaded - so something like `export DIRECT_URL="..."`. You could probably undo that via running `export DIRECT_URL=""`.

As long as `db push` displays a port different than the one that is defined in `.env` as `DIRECT_URL` something is off with your setup.

> FYI: I sent you a DM (andreasobi) on discord with the connection strings.

I received and approved a friend request, but no data. Can you send again?

**Comment by chadnpc** [comment]
Hey @andreasobi ,
you're saying it was working a few days ago but idk the error msg `Can't reach database server` is not specific
troubleshooting this will be quite different for every person.

 There are multiple things that could have caused this:
ex: *database server* (supabase) *was paused* or *you changed something* (like reset ur password)

**Fix**: 
1. update prisma
  ```bash
  npm i --save-dev prisma@latest
  ```
2. Edit your .env files (local or production idk)
  - Replace any special characters in your db password strings with their [html urlencoding](https://www.w3schools.com/tags/ref_urlencode.ASP). in DATABASE_URL and DIRECT_URL If there are any.

  - The port 6543 is typically used for pgBouncer, which can cause issues with Prisma migrations. For migrations, I just use the direct connection to the database. so remove it! (It worked for me 🤷🏻‍♂️)
     so the .env will look like:
    >DATABASE_URL="postgres://postgres.xjetlyrodrrupwokmxsa:{urlsafe-password}@aws-0-eu-central-1.pooler.supabase.com:6543/postgres"
    DIRECT_URL="postgres://postgres.xjetlyrodrrupwokmxsa:{urlsafe-password}@aws-0-eu-central-1.pooler.supabase.com:5432/postgres"
    
      Note: *change the DATABASE_URL port to 5432. If your database is Postgres 14*

3. [Resume](https://github.com/orgs/supabase/discussions/20377) you database server and test if connection goes through:

  ```bash
  psql "postgresql://postgres.xjetlyrodrrupwokmxsa:{urlsafe-password}@aws-0-eu-central-1.pooler.supabase.com:6543"
  ```
  4. If everything is Ok, run
  ```bash
  npx prisma migrate dev
  ```
  Thats it!
  
I hope this helps.

**Reply by AbdullahAjayi** [reply]
Option 2 helped! Thanks a ton.

**Comment by Patrique-x86** [comment]
Hey @andreasobi I'm encountering the same problem. I realized I forgot to replace the accurate Supabase PostgreSQL password in the connection string, so double-check it. (.env)

**Reply by rishabhr4** [reply]
I double checked it and also changed the password as it was mentioned is some discussions

**Reply by rishabhr4** [reply]
Still not working

**Comment by nihadtalibzada** [comment]
Having the same issue here...

**Reply by rishabhr4** [reply]
Use npx prisma generate and then check 
Worked for me

**Reply by nihadtalibzada** [reply]
I already generated it, but I tried again. Still no success. DB connection is failing

**Reply by nihadtalibzada** [reply]
Oh man, I was using my account password instead of DB password 🤦🏻‍♂️ 

So, make sure it's your DB password in your .env file.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;

**Comment by engalisabry** [comment]
## How I Solved Prisma Migration Error with Supabase (P1001)

After a lot of trial and error — deleting and recreating the Supabase project, changing the database password, and attempting migrations multiple times — I finally found a solution.

In my case, I was using **Prisma with Supabase**.

- Testing the database connection manually using `psql` **worked fine**.
- Running a Prisma migration **kept failing** with a connection error (P1001).

---

###  Solution

I added **timeout parameters** to both `DATABASE_URL` and `DIRECT_URL` in my `.env` file.

---

### ❌ Old Version (was failing)

```env
# Pooled connection for Prisma Client
DATABASE_URL="postgresql://postgres:[YOUR_PASSWORD]@aws-0-us-east-2.pooler.supabase.com:5432/postgres?pgbouncer=true"

# Direct connection for Prisma Migrate
DIRECT_URL="postgresql://postgres:[YOUR_PASSWORD]@db.[PROJECT_REF].supabase.co:5432/postgres"

```
### ✅ New Version (worked for me)


```env
# Pooled connection for Prisma Client
DATABASE_URL="postgresql://postgres:[YOUR_PASSWORD]@aws-0-us-east-2.pooler.supabase.com:5432/postgres?pgbouncer=true&connect_timeout=30&pool_timeout=30"

# Direct connection for Prisma Migrate
DIRECT_URL="postgresql://postgres:[YOUR_PASSWORD]@db.[PROJECT_REF].supabase.co:5432/postgres?connect_timeout=30"

```

Hope this helps someone else struggling with the same issue! 🚀

**Reply by thisisjohnyenglish** [reply]
That worked! thanks

**Reply by mrrobotke** [reply]
@engalisabry solution should be pinned, worked like a charm
